### PR TITLE
Update Documentation: Bots Architeture

### LINKED ISSUE #14552: Documentation: revise Bots architecture
As of now, the [architecture](https://rocket.chat/docs/bots/bots-architecture/) page contains information about constituent parts but for a person who wants to get himself familiar with, it is hard to understand how these parts work together on the high level.

### PR DESCRIPTION
**Pull request by bandantonio**
Revised bots architecture page:
* Added more context to make working principles more clear
* Added anchors to the fundamental architecture elements
* Removed time-dependent information
* Added framework scalability to the bot architectural diagram
* Added section describing the importance of scalability

Closes https://github.com/RocketChat/Rocket.Chat/issues/14552

### CODE REVIEW

**Review by Sing-Li — COMMENTED** [review]
This is excellent!

We didn't even get a chance to talk yet.  It is already 80%.  Let's talk briefly soon.

**Review by Sing-Li — COMMENTED** [review]
Please review changes.

File: bots/bots-architecture/README.md [hunk 1]
```diff
@@ -37,31 +50,52 @@ emitted by the server's streamer. Bots subscribe to the collection of messages
...
 
-Bot frameworks such as Hubot, Botkit, Rasa and Botpress each provide unique
-approaches to composing conversational interfaces. Some run their own hosting
-for bots, some just provide the code for you to run yourself. Due to the magic
-of adapters, our architecture is mostly independent to your choice of framework.
+Bot frameworks (such as Hubot, Botkit, Rasa or Botpress) provide unique
+approaches to create conversational interfaces and allow developers
+to focus on the design and management of complex bot logic.
+
+Most of framworks can be run on your own bot host. There are also frameworks
```
Comment (Sing-Li) [inline bots/bots-architecture/README.md:78]: Not true.   Actually they operate as a business on the cloud, charging monthly fees.   This needs to be re-worded.    These frameworks are platforms and may not be "distributed".

File: bots/bots-architecture/README.md [hunk 2]
```diff
@@ -37,31 +50,52 @@ emitted by the server's streamer. Bots subscribe to the collection of messages
...
+
+Most of the popular frameworks have adapters created by the Rocket.Chat's community.
+
+## Bot Scaling
+
+In production, hundreds or thousands of users can be carrying out conversations with a bot
+at the same time. The ability to scale a bot from one conversation to many concurrent ones
+is essential.
+
+Bot frameworks/platforms support varying degrees of scalability. To ensure that growing amount
```
Comment (Sing-Li) [inline bots/bots-architecture/README.md]: growing *number* of users?

Comment (bandantonio) [reply bots/bots-architecture/README.md]: Definitely

File: bots/bots-architecture/README.md (hunk 1 above)
Comment (bandantonio) [reply bots/bots-architecture/README.md:78]: Fixed. Changed to "operate"

**Review by Sing-Li — APPROVED** [review]

**Pull request merged**